tests/nlu/test_components.py

```diff
@@ -200,3 +200,17 @@ async def test_validate_requirements_raises_exception_on_component_without_name(
         await train(
             _config, data="./data/examples/rasa/demo-rasa.json", path=str(tmp_path),
         )
+
+
+async def test_validate_component_keys_raises_warning_on_invalid_key(tmp_path: Path,):
+    _config = RasaNLUModelConfig(
+        # config with a component that does not have a `name` property
+        {"pipeline": [{"name": "WhitespaceTokenizer", "confidence_threshold": 0.7}]}
```

Comment:
wait, isn't `name` the thing which shouldn't be checked? You're even explicitly excluding that in your code, no?

Reply:
Name has to be provided so that `validate_requirements` doesn't raise an exception, and so that the component class can be fetched

Reply:
https://github.com/RasaHQ/rasa/pull/7134/files#r516065782